Fixes: Bug 23595729 dnet can't send out packet

This fix is outdated and this code is not called for a long time because
eth-bsd.c is selected as an eth module during configure time instead of
eth-dlpi.c.

I am keeping this patch here just to note that the code in eth-dlpi.c is broken
on Solaris.

Not for upstream.

--- a/src/eth-dlpi.c	2023-10-12 18:15:10.000000000 +0200
+++ b/src/eth-dlpi.c	2024-02-07 08:53:24.010770707 +0100
@@ -151,7 +151,7 @@
 	}
 #else
 	e->fd = -1;
-	snprintf(dev, sizeof(dev), "/dev/%s", device);
+	snprintf(dev, sizeof(dev), "/dev/net/%s", device);
 	if ((p = dev_find_ppa(dev)) == NULL) {
 		errno = EINVAL;
 		return (eth_close(e));
@@ -160,7 +160,7 @@
 	*p = '\0';
 
 	if ((e->fd = open(dev, O_RDWR)) < 0) {
-		snprintf(dev, sizeof(dev), "/dev/%s", device);
+		snprintf(dev, sizeof(dev), "/dev/net/%s", device);
 		if ((e->fd = open(dev, O_RDWR)) < 0)
 			return (eth_close(e));
 	}
